prod.Dockerfile

```diff
@@ -0,0 +1,57 @@
+FROM node:18-bookworm as base_builder
+
+WORKDIR /usr/src/app
+
+RUN npm install -g pnpm
+COPY pnpm-lock.yaml .
+RUN pnpm fetch
+
+COPY . .
+RUN pnpm install -f --offline
+
+FROM base_builder as backend
+WORKDIR /usr/src/app/packages/hoppscotch-backend
+RUN pnpm exec prisma generate
+RUN pnpm run build
+CMD ["pnpm", "run", "start:prod"]
+EXPOSE 3170
+
+FROM base_builder as fe_builder
+WORKDIR /usr/src/app/packages/hoppscotch-selfhost-web
+RUN pnpm run generate
+
+FROM caddy:2-alpine as app
+WORKDIR /site
+COPY --from=fe_builder /usr/src/app/packages/hoppscotch-sh-admin/prod_run.mjs /usr
+COPY --from=fe_builder /usr/src/app/packages/hoppscotch-selfhost-web/Caddyfile /etc/caddy/Caddyfile
+COPY --from=fe_builder /usr/src/app/packages/hoppscotch-selfhost-web/dist/ .
+RUN apk add nodejs npm
+RUN npm install -g @import-meta-env/cli
+EXPOSE 8080
+CMD ["/bin/sh", "-c", "node /usr/prod_run.mjs && caddy run --config /etc/caddy/Caddyfile --adapter caddyfile"]
+
+FROM base_builder as sh_admin_builder
+WORKDIR /usr/src/app/packages/hoppscotch-sh-admin
+RUN pnpm run build
+
+FROM caddy:2-alpine as sh_admin
+WORKDIR /site
+COPY --from=sh_admin_builder /usr/src/app/packages/hoppscotch-sh-admin/prod_run.mjs /usr
+COPY --from=sh_admin_builder /usr/src/app/packages/hoppscotch-sh-admin/Caddyfile /etc/caddy/Caddyfile
+COPY --from=sh_admin_builder /usr/src/app/packages/hoppscotch-sh-admin/dist/ .
+RUN apk add nodejs npm
+RUN npm install -g @import-meta-env/cli
+EXPOSE 8080
+CMD ["/bin/sh", "-c", "node /usr/prod_run.mjs && caddy run --config /etc/caddy/Caddyfile --adapter caddyfile"]
+
+FROM backend as aio
+RUN apt-get update
+RUN apt-get install -y caddy
+RUN npm install -g @import-meta-env/cli
+COPY --from=fe_builder /usr/src/app/packages/hoppscotch-selfhost-web/dist /site/selfhost-web
+COPY --from=sh_admin_builder /usr/src/app/packages/hoppscotch-sh-admin/dist /site/sh-admin
+COPY aio.Caddyfile /etc/caddy/Caddyfile
```

Comment:
Maybe we can add a healthcheck before `CMD` to check if all our services are functioning properly. We could have a script we run that checks everything and call that script in the `HEALTHCHECK` stage.